Delay hash change during file export on iOS 13.x+

nibhar opened the pull request:
While Safari on iOS 13 does support the download attribute, the resulting download prompt does not like the hash change  in the background happening in our setup. 
This PR changes that behaviour by delaying the hash change until after the prompt was confirmed.

This PR also increases core-web to Version 1.5.3.

Resolves #382

nibhar committed:
core-web version update to 1.5.3

nibhar committed:
add a confirmation step during file export for iOS 13.x+

sisou commented on src/components/DownloadLoginFile.js:
```diff
@@ -166,8 +177,16 @@ class DownloadLoginFile extends Nimiq.Observable {
             // consider the download successful.
-            if (!document.hasFocus()) {
+            // As mobile Safari on iOS 13.x+ does support the download attribute, it no longer uses the long tap
+            // fallback. However, if the page changes its hash in the background while the promt asking to download
```
Typo `promt => prompt`

sisou reviewed (approved):
Perfect! I'll add the Nimiq lib hashes in a few minutes ;).

sisou committed:
Update integrity hashes for Nimiq lib v1.5.3

nibhar committed:
download again button now triggers download

download file screen always in initial state, also after closing the following success screen.

danimoh commented on src/components/DownloadLoginFile.js:
```diff
@@ -42,7 +46,17 @@ class DownloadLoginFile extends Nimiq.Observable {
+        });
+        $backToDownloadButton.addEventListener('click', () => {
+            this.$downloadButton.click();
+        });
```
No need to open a block here.

danimoh commented on src/components/DownloadLoginFile.js:
```diff
@@ -90,6 +105,9 @@ class DownloadLoginFile extends Nimiq.Observable {
         }
+        // Remove previously added classes to restore the initial state.
+        this.$el.classList.remove('maybe-downloaded');
+        this.fire(DownloadLoginFile.Events.RESET);
```
You're using this twice. Might be worth moving this to a `reset` method.

danimoh commented on src/components/DownloadLoginFile.js:
```diff
@@ -166,8 +184,16 @@ class DownloadLoginFile extends Nimiq.Observable {
+            if (!document.hasFocus()
+                || (BrowserDetection.isIOS()
+                    && BrowserDetection.isSafari()
+                    && BrowserDetection.iOSVersion()[0] > 12)) {
```
I think we should test whether we need to include iOS Chrome too.
Or didn't we even say we want to have the `Continue` button always on mobile?

nibhar replied:
Actually I think we should probably just get rid of the `BrowserDetection.isSafari()` condition.
For non iOS devices I think the continue button is simply not necessary, but on iOS 13 we can just declare this as default behaviour, no matter what browser.

nibhar committed:
All browsers on iOS 13+ share the download file functionality

danimoh reviewed (approved):
Resolves https://github.com/nimiq/keyguard/issues/382
